classes/local/store/object_file_system.php

```diff
@@ -316,6 +321,50 @@ public function delete_object_from_local_by_hash($contenthash, $objectsize = 0)
         return $finallocation;
     }
 
+    /**
+     * @param string $rootpath
+     */
+    public function delete_empty_folders($rootpath) {
+        if (!is_dir($rootpath)) {
+            return;
+        }
+        if ($this->is_dir_empty($rootpath)) {
+            rmdir($rootpath);
+            return;
+        }
+        $iterator = new RecursiveDirectoryIterator($rootpath);
```

Comment:
It's not clear from the docs if RecursiveDirectoryIterator is top down or bottom up. It must really be bottom up, and if it is then it should in theory delete the root itself last without special handling.

I'd like to see unit tests around this